src/main/java/org/entando/kubernetes/service/digitalexchange/BundleUtilities.java

```diff
@@ -72,10 +73,13 @@ public static String extractIngressPathFromDescriptor(PluginDescriptor descripto
     }
 
     private static String composeNameFromDockerImage(DockerImage image) {
-        return String.join("-",
+        String name = String.join("-",
                 makeKubernetesCompatible(image.getOrganization()),
                 makeKubernetesCompatible(image.getName()),
                 makeKubernetesCompatible(image.getVersion()));
+
+        return name.substring(0, Math.min(33, name.length())) // final string has not be longer than 63 chars
```

Comment:
Wonder if a silent truncate is safe.
Is the plugin name supposed to be unique?

Reply:
This is an example of a pod name:
`lcorsettientando-the-lucas-0-0-1-server-deployment-58d859dpd5fp`

the segment `-server-deployment-58d859dpd5fp` is autogenerated by k8s and appended to the first part. length 30

Max accepted length is 63, so the segment created by us must have a max length = 33
This is the segment is created by us: `lcorsettientando-the-lucas-0-0-1` max length 33 (trimming the final hyphen to avoid a double -- that is not compatible)

Thanks to the k8s suffix we should be safe about name collisions, but we may lose the docker image version in the pod name, as you can see in the example above

Reply:
> Max accepted length is 63, so the segment created by us must have a max length = 32

Please make it explicit on code.
Use constants and make explicit calculation.

> Thanks to the k8s suffix we should be safe about name collisions, but we may lose the docker image version in the pod name, as you can see in the example above

right, but still I'm not happy having something in the descriptor and something else installed.
I'd say that the installation should fail in this case.
We may try to reduce the possibility to trigger it by doing some on the CLIs

Reply:
maybe we can introduce a validation step that verifies the max length of the image field?

Reply:
and I may move the constants in the validator

Reply:
@ampie @joewhite101 @ffleandro thougts?

Reply:
@ampie do you think it's possible to set a max-len limit at CRD level on this field?